### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by olya-yevsieienko**

### CODE REVIEW

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorOldFormat = fromFormat[3];
+  const separatorNewFormat = toFormat[3];
+
+  const oldFormat = date.split(separatorOldFormat);
```
Comment (Olexandr6) [inline src/formatDate.js:56]:
```suggestion
  const oldDateFormat = date.split(separatorOldFormat);
```

File: src/formatDate.js [hunk 2]
```diff
@@ -50,7 +50,29 @@
...
+  const newFormat = [];
+
+  for (let i = 0; i < oldFormat.length; i++) {
+    if (fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0) {
```
Comment (Olexandr6) [inline src/formatDate.js:60]: For such long condition better to create a separate variable to describe what you expect
```suggestion
		const isDifferentYearFormat = fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0;

    if (isDifferentYearFormat) {
```

**Review by Olexandr6 — CHANGES_REQUESTED** [review]
Your solution is cool🔥 Let's work on some improvements

**Comment by olya-yevsieienko** [comment]
Okey, I got it. Thanks!

вт, 14 июн. 2022 г. в 13:39, Olexandr Yatsykovych ***@***.***

File: src/formatDate.js [hunk 3]
```diff
@@ -50,7 +50,32 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatoroldDateFormat = fromFormat[3];
```
Comment (Olexandr6) [inline src/formatDate.js:53]:
```suggestion
  const separatorOldDateFormat = fromFormat[3];
```

File: src/formatDate.js [hunk 4]
```diff
@@ -50,7 +50,32 @@
...
+  const separatornewDateFormat = toFormat[3];
```
Comment (Olexandr6) [inline src/formatDate.js:54]:
```suggestion
  const separatorNewDateFormat = toFormat[3];
```

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**